apps/query-ui/queryui/tests/integration/Chat_test.py

```diff
@@ -0,0 +1,276 @@
+        "<TestBlock top>:0",
+        "<TestBlock a>:1",
+        "<TestBlock b>:2",
+        "<TestBlock b1>:3",
+        "<TestBlock b2>:4",
+        "<TestBlock c>:5",
+    ]
+
+
+def test_expect_any(block_tree):
+    """Test the expect_any function."""
+    top = block_tree
+    expect_any(top, lambda n: n.label == "a")
+    expect_any(top, lambda n: n.label == "b")
+    expect_any(top, lambda n: n.label == "c")
+    expect_any(top, lambda n: n.label == "a" or n.label == "b2")
+    with pytest.raises(AssertionError):
+        expect_any(top, lambda n: n.label == "d")
+
+
+def test_expect_all(block_tree):
+    """Test the expect_all function."""
+    top = block_tree
+    expect_all(top, lambda n: n.label)
+    with pytest.raises(AssertionError):
+        expect_all(top, lambda n: n.label == "a")
+
+
+def test_expect_none(block_tree):
+    """Test the expect_none function."""
+    top = block_tree
+    expect_none(top, lambda n: n.label == "d")
+    with pytest.raises(AssertionError):
+        expect_none(top, lambda n: n.label == "a")
+
+
+def test_app_starts(app):
+    """Test that the app starts without raising an exception."""
+    assert app.title[0].value == "Sycamore Query Chat"
+    assert not app.exception
+
+
+def check_query_result(result: ChatMessage) -> Tuple[str, str]:
+    """Check that the query result satisfies the expected format.
+
+    Returns the markdown content of the result text and the query traces.
+    """
+
+    # Should be no exception nodes.
+    expect_none(result, lambda x: x.type == "exception")
+
+    # Query plan should be present.
+    expanders: List[Expander] = [
+        x for x in result.children.values() if x.type == "expander" and isinstance(x, Expander)
+    ]
+    # assert all([isinstance(x, Expander) for x in expanders])
+    assert expanders[0].label == "Query plan"
+    assert expanders[0].children[0].type == "json"
+    assert isinstance(expanders[0].children[0], Json)
+    query_plan_json = json.loads(expanders[0].children[0].value)
+    assert isinstance(query_plan_json, dict)
+    assert "query" in query_plan_json
+    assert set(query_plan_json.keys()) == {"query", "nodes", "result_node", "llm_prompt", "llm_plan"}
+
+    # Sycamore query result should be present.
+    assert expanders[1].label == "Sycamore query result"
+    assert expanders[1].children[0].type == "markdown"
+
+    # Query traces should be present.
+    assert expanders[2].label == "Query trace"
+    print(expanders[2])
+
+    assert expanders[2].children[0].type == "vertical"
+    assert expanders[2].children[0].children[0].children[0].label == "Node data"  # type: ignore
+    assert expanders[2].children[0].children[0].children[1].label == "Query plan"  # type: ignore
+    assert expanders[2].children[0].children[0].children[0].children[1].type == "markdown"
+    return get_markdown(result), get_markdown(expanders[2].children[0].children[0].children[0])
+
+
+def test_count_query(app):
+    """Test a basic count query."""
```

Comment:
Do we want to test an LLM operator (LLM query) based query as well? It is a different type of operator that could easily break as well.

Reply:
Yep, I want to add a bunch more tests once the basic framework is landed!